components/core/src/clp_s/TimestampDictionaryWriter.hpp

```diff
@@ -91,14 +78,24 @@ class TimestampDictionaryWriter {
      */
     epochtime_t get_end_timestamp() const;
 
+    /**
+     * Clears and resets all internal state.
+     */
+    void clear();
+
+    /**
+     * Merge ranges by key name then return the size of data to be compressed in bytes
```

Comment:
Not sure if we should do two different things in one function. Additionally, directly adding the sizes of different types may not be intuitive to readers.

Reply:
Yeah, the unfortunate thing is that because of how this class is written you always need to merge ranges before you can determine the size of all of the structures that get stored.

One thing we could do is get rid of this function and instead change the `write` function to write to a buffer. That way we can just look at the final size of the buffer to get the pre-zstd size. The timestamp dictionary shouldn't be more than a few hundred bytes, so writing it to a buffer should be fine.